static/js/search.js

```diff
@@ -220,6 +220,7 @@ export function focus_search() {
 }
 
 export function initiate_search() {
+    $("#search_query").val("");
     message_view_header.open_search_bar_and_close_narrow_description();
```

Comment:
I couldn't find a bug that this change would cause, but feels like this could mess with the search pills code. So, I suggest we do `$("#search_query").val("")` in `recent_topics_ui.show`. @YashRE42 FYI if you have time to review.